File: src/priv/bajdcc/LALR1/grammar/runtime/IRuntimeStack.java
package priv.bajdcc.LALR1.grammar.runtime;

/**
 * 运行时堆栈接口
 *
 * @author bajdcc
 */
public interface IRuntimeStack {

	RuntimeObject load() throws RuntimeException;
	void store(RuntimeObject obj);
	void push() throws RuntimeException;
	void pop() throws RuntimeException;
	void opLoad() throws RuntimeException;
	void opLoadFunc() throws RuntimeException;
	void opStore() throws RuntimeException;
	void opStoreDirect() throws RuntimeException;
	void opOpenFunc() throws RuntimeException;
	void opLoadArgs() throws RuntimeException;
	void opPushArgs() throws RuntimeException;
	void opReturn() throws RuntimeException;
	void opCall() throws RuntimeException;
	void opPushNull();
	void opPushZero();
	void opPushNan();
	void opLoadVar() throws RuntimeException;
	void opJump() throws RuntimeException;
	void opJumpBool(boolean bool) throws RuntimeException;
	void opJumpBoolRetain(boolean bool) throws RuntimeException;
	void opJumpZero(boolean bool) throws RuntimeException;
	void opJumpYield() throws RuntimeException;
	void opJumpNan() throws RuntimeException;
	void opImport() throws RuntimeException;
	void opLoadExtern() throws RuntimeException;
	void opCallExtern(boolean invoke) throws Exception;
	void opYield(boolean input) throws RuntimeException;
	void opYieldSwitch(boolean forward) throws RuntimeException;
	void opYieldCreateContext() throws Exception;
	void opYieldDestroyContext() throws RuntimeException;
	void opScope(boolean enter) throws RuntimeException;
	void opArr();
	void opMap();
}

File: src/main/kotlin/com/bajdcc/LALR1/grammar/runtime/IRuntimeStack.kt
package com.bajdcc.LALR1.grammar.runtime

/**
 * 运行时堆栈接口
 *
 * @author bajdcc
 */
interface IRuntimeStack {

    @Throws(RuntimeException::class)
    fun load(): RuntimeObject

    @Throws(RuntimeException::class)
    fun store(obj: RuntimeObject)

    @Throws(RuntimeException::class)
    fun push()

    @Throws(RuntimeException::class)
    fun pop()

    @Throws(RuntimeException::class)
    fun opLoad()

    @Throws(RuntimeException::class)
    fun opLoadFunc()

    @Throws(RuntimeException::class)
    fun opReloadFunc()

    @Throws(RuntimeException::class)
    fun opStore()

    @Throws(RuntimeException::class)
    fun opStoreCopy()

    @Throws(RuntimeException::class)
    fun opStoreDirect()

    @Throws(RuntimeException::class)
    fun opOpenFunc()

    @Throws(RuntimeException::class)
    fun opLoadArgs()

    @Throws(RuntimeException::class)
    fun opPushArgs()

    @Throws(RuntimeException::class)
    fun opReturn()

    @Throws(RuntimeException::class)
    fun opCall()

    @Throws(RuntimeException::class)
    fun opPushNull()

    @Throws(RuntimeException::class)
    fun opPushZero()

    @Throws(RuntimeException::class)
    fun opPushNan()

    @Throws(RuntimeException::class)
    fun opPushPtr(pc: Int)

    @Throws(RuntimeException::class)
    fun opPushObj(obj: RuntimeObject)

    @Throws(RuntimeException::class)
    fun opLoadVar()

    @Throws(RuntimeException::class)
    fun opJump()

    @Throws(RuntimeException::class)
    fun opJumpBool(bool: Boolean)

    @Throws(RuntimeException::class)
    fun opJumpBoolRetain(bool: Boolean)

    @Throws(RuntimeException::class)
    fun opJumpZero(bool: Boolean)

    @Throws(RuntimeException::class)
    fun opJumpYield()

    @Throws(RuntimeException::class)
    fun opJumpNan()

    @Throws(RuntimeException::class)
    fun opImport()

    @Throws(RuntimeException::class)
    fun opLoadExtern()

    @Throws(Exception::class)
    fun opCallExtern(invoke: Boolean)

    @Throws(RuntimeException::class)
    fun opYield(input: Boolean)

    @Throws(RuntimeException::class)
    fun opYieldSwitch(forward: Boolean)

    @Throws(Exception::class)
    fun opYieldCreateContext()

    @Throws(RuntimeException::class)
    fun opYieldDestroyContext()

    @Throws(RuntimeException::class)
    fun opScope(enter: Boolean)

    @Throws(RuntimeException::class)
    fun opArr()

    @Throws(RuntimeException::class)
    fun opMap()

    @Throws(RuntimeException::class)
    fun opIndex()

    @Throws(RuntimeException::class)
    fun opIndexAssign()

    @Throws(RuntimeException::class)
    fun opTry()

    @Throws(RuntimeException::class)
    fun opThrow()
}